config/env/local_client-external.json

```diff
@@ -0,0 +1,23 @@
+{
+  "services": {
+    "ceramic": {
+      "launchLocation": "serviceLauncher",
+      "mode": "client",
+      "apiURL": "http://localhost:7007",
+      "port": 7007
```

Comment:
why is port needed if we're connecting with the http client and have `apiURL` already?

EDIT: I see now.  I think my confusion is that we're grouping configuration designed for consumption in two entirely different places together.  Could we try separating out two different top-level config objects, something like "testServices" and "preLaunchedServices" (not great names, feel free to propose alternatives), keeping the config for how the jest tests set up / look for existing services from how the service launcher launches services in advance.  The service launcher config doesn't even need to have specific names for these things (i.e. it doesn't care about which one is "ceramic" and which one is "ceramic2"), it could just take an array of things to launch.

Reply:
The best way to do this is to break up the url and have
```js
  ceramic: {
    host: 'localhost'
    scheme: 'http'
    port: 7007
  },
```

Then reconstruct the pieces however you need them. In our case using localhost we have port in the url but this may not always be the case (for example if the server uses a proxy--in this case we don't but the point is the config should be explicit and reduce duplication/aka human error as much as possible).

We still do need ceramic2, 3, etc in json so we know how many things to launch and what the unique config is for each of them.

Reply:
I'm not really convinced.  These are conceptually completely separate configs.  The fact that they live in the same config file is total happenstance based on the way we've built the "service launcher", just for convenience sake.  But the service launcher could just as easily be a python or bash script, with its entirely own configuration.  It serves a totally different role and is used at a totally different time during the test process.  By the time jest has started running the actual tests, the configs for the service launcher are now completely irrelevant, all that matters is that there are some ceramic/ipfs process running where the tests expect them to be running.  Frankly I'd say put the configs in completely different files if it wouldn't be a pain to do so.  We did this all in javascript just because that's what we as a team are most familiar with and so it was convenient to do so, but keeping these configs in the same place risks confusion as now when writing a new test I worry that it's not clear what configs control the test behavior and what configs control the behavior of the pre-tests setup process.  I think separating these configs out from each other as much as possible will make things clearer.

Feel free to push back though if I've failed to convince you!  To me it seems clear that keeping these configs separate will make things easier to understand but maybe that's not necessarily true for everyone.  I just see these other comments around the roles and intersections of things like `mode` and `launchLocation` and that seems like a red flag to me that we've conflated two concepts into one place that would be better served being separate

Reply:
Yeah I think it’s fine to split it up to avoid confusion